x-pack/metricbeat/module/azure/billing/billing.go

```diff
@@ -34,11 +36,36 @@ type MetricSet struct {
 
 // Config options
 type Config struct {
-	ClientId       string        `config:"client_id"    validate:"required"`
-	ClientSecret   string        `config:"client_secret" validate:"required"`
-	TenantId       string        `config:"tenant_id" validate:"required"`
-	SubscriptionId string        `config:"subscription_id" validate:"required"`
-	Period         time.Duration `config:"period" validate:"nonzero,required"`
+	ClientId                string        `config:"client_id"    validate:"required"`
+	ClientSecret            string        `config:"client_secret" validate:"required"`
+	TenantId                string        `config:"tenant_id" validate:"required"`
+	SubscriptionId          string        `config:"subscription_id" validate:"required"`
+	Period                  time.Duration `config:"period" validate:"nonzero,required"`
+	ResourceManagerEndpoint string        `config:"resource_manager_endpoint"`
+	ActiveDirectoryEndpoint string        `config:"active_directory_endpoint"`
+}
+
+func (conf *Config) Validate() error {
+	if conf.ResourceManagerEndpoint == "" {
+		conf.ResourceManagerEndpoint = azure.DefaultBaseURI
+	}
+	if conf.ActiveDirectoryEndpoint == "" {
```

Comment:
Same code seems to be used in `x-pack/metricbeat/module/azure/config.go`? Should this be in a function that can be commonly used in both places?

Reply:
moved to one config, had to change some logic there.